src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -253,8 +248,15 @@ class DeepDiveLogParser extends JavaTokenParsers {
       }
     )
 
-  def functionMode = "mode" ~> "=" ~> functionModeType
-  def inferenceMode = "mode" ~> "=" ~> inferenceModeType
+  def ruleAnnotation = "@" ~> annotationName ~ ("(" ~> rep1sep(ident, ",") <~ ")") ^^ {
+    case (name ~ args) => RuleAnnotation(name, args)
+  }
+
+  def ruleAnnotations = rep(ruleAnnotation)
```

Comment:
Okay, sounds good.  So #20 stays open.  When doing this later, let's export the annotations on the rules into JSON as well.  That means we probably need to come up with a nice and compact way to represent the parse tree for rules in JSON first.